impeller/entity/entity_pass.cc

```diff
@@ -48,6 +48,11 @@ void EntityPass::SetDelegate(std::unique_ptr<EntityPassDelegate> delegate) {
 }
 
 void EntityPass::AddEntity(Entity entity) {
+  if (entity.GetBlendMode() == BlendMode::kSourceOver &&
+      entity.GetContents()->IsOpaque()) {
+    entity.SetBlendMode(BlendMode::kSource);
+  }
```

Comment:
Does this happen before or after we do the opacity peephole check?

Reply:
This happens before (when Aiks pushes an Entity into the pass).

Hmm, it looks like we weren't handling blend modes in the opacity peephole. I added some restrictions and made it so it's able to unfold things with the Source blend mode. There's a new utility where the `Entity` handles `SetInheritedOpacity` with the blend mode in mind. Working on test cases for this at the moment.

Reply:
Like if the pass has kLuminosity as its blend mode and the child entities have kHue, it'll incorrectly collapse the pass.

Reply:
Ah nvm about the pass. I think the entities are still not being checked right, though.